ocaml/util/gen_build_info_ml.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -e
+
+DATE=$(date "+%Y-%m-%d")
+# shellcheck disable=SC2001
+XAPI_VERSION=$(echo "$XAPI_VERSION" | sed "s/^v//g")
+
+printf "let date = \"%s\"\n\n" "$DATE"
```

Comment:
Please do no add the date to the executable as part of the build process. It breaks reproducible builds.

Reply:
We've already got that. It was just done in the dune file itself.

Reply:
this was already there:

```
  (action (with-stdout-to %{targets} (run "date" "+let date=\"%Y-%m-%d\"")))
```
it gets used here:

```
let software_version () =
  (* In the case of XCP, all product_* fields will be blank. *)
  List.filter (fun (_, value) -> value <> "")
    [_product_version, Xapi_version.product_version ();
...
     _date,            Xapi_version.date]
```

Reply:
Let's leave this for another PR.